components/draw/GeometryMode.svelte

```diff
@@ -0,0 +1,156 @@
+  export let snapTool;
+  export let drawControls;
+
+  $: {
+    if (mode == "edit-geometry") {
+      drawControls.changeMode("static");
+    }
+  }
+
+  // This is different than the stores.js one!
+  let currentlyEditing = null;
+
+  // Calculate hover
+  $map.on("mousemove", (e) => {
+    if (mode == "edit-geometry" && currentlyEditing == null) {
+      let results = $map.queryRenderedFeatures(e.point, {
+        layers: [
+          "interventions-points",
+          "interventions-lines",
+          "interventions-polygons",
+        ],
+      });
+      // TODO ? syntax
+      var newHoverId = null;
+      if (results.length > 0) {
+        newHoverId = results[0].id;
+      }
+      currentHover.set(newHoverId);
+    }
+  });
+  $map.on("mouseout", () => {
+    if (mode == "edit-geometry" && currentlyEditing == null) {
+      currentHover.set(null);
+    }
+  });
+
+  // Handle clicking the hovered feature
+  $map.on("click", (e) => {
+    if (mode == "edit-geometry" && currentlyEditing == null) {
+      let results = $map.queryRenderedFeatures(e.point, {
+        layers: [
+          "interventions-points",
+          "interventions-lines",
+          "interventions-polygons",
+        ],
+      });
+      if (results.length > 0) {
+        startEditing(results[0].id);
+      }
+    }
+  });
+
+  snapTool.addEventListener("new-route", (e) => {
+    if (mode == "edit-geometry") {
+      const editedRoute = e.detail;
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        // Keep the ID and any properties. Just copy over stuff from routeSnapper.
+        // TODO We're depending on implementation details here and knowing what to copy...
+        feature.properties.length_meters = editedRoute.properties.length_meters;
+        feature.properties.waypoints = editedRoute.properties.waypoints;
+        delete feature.properties.hide_while_editing;
+        feature.geometry = editedRoute.geometry;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+
+  $map.on("draw.update", (e) => {
+    if (mode == "edit-geometry") {
+      // Assume there's exactly 1 feature
+      const feature = e.features[0];
+
+      gjScheme.update((gj) => {
+        let updateFeature = gj.features.find((f) => f.id == currentlyEditing);
+        updateFeature.geometry = feature.geometry;
+        return gj;
+      });
+
+      // Stay in this mode
+    }
+  });
+
+  $map.on("draw.selectionchange", (e) => {
+    if (mode == "edit-geometry" && e.features.length == 0) {
+      drawControls.changeMode("static");
+      gjScheme.update((gj) => {
+        let feature = gj.features.find((f) => f.id == currentlyEditing);
+        delete feature.properties.hide_while_editing;
+        return gj;
+      });
+
+      // Stay in this mode
+      currentlyEditing = null;
+    }
+  });
+
+  function startEditing(id) {
+    currentHover.set(null);
```

Comment:
Maybe it's confusing to abuse the hover layer here. It makes entries in the sidebar light up too. (But... that could be nice; people need to know the name of the thing they're about to edit.) But... then interacting with sidebar entries while in this mode gets confusing.

Reply:
I think this is fine, although note the UX you're describing would be nice to have.